enhancements/bundle-validation.md

```diff
@@ -0,0 +1,221 @@
+
+- **Issue 2:** Rather than rewriting the courier project to provide static analysis for bundles provided as Catalog Images, the new library can be written and then imported by tools like _scorecard_ to perform validation.
+
+### Goals
+
+Define a set of rules/validations so that bundles with their specific set of configurations can be validated by checking the required information and components.
+
+### Non-Goals
+- The validation compares different versions of bundles from the same package.
+ 
+- The validation considers the validity of a bundle in the context of an index, which should be part of [operator-registry](https://github.com/operator-framework/operator-registry).
+
+- The library includes dynamic bundle analysis on bundles with respect to specific cluster settings.
+
+- The validation guarantees the successful installation of a bundle.
+
+## Proposal
+
+### User Stories
+
+As an Operator developer or maintainer, I would like to discover my missing vital metadata and validate that my Operator bundle is well-formed. 
+
+As a cluster administrator or release pipeline owner, I would want to be aware of malformed Operator bundles so that I can prevent them from being released into a catalog for cluster tenants to install.
+
+As a community contributor to the bundle validation library, I would want to understand the workflow of contributing to the core validation strategies or customized validations as part of an extended package.
+
+### Implementation Details/Notes/Constraints
+- The bundle validation library should list out errors based on static scanning of a bundle directory and locate their root causes if applicable. 
+
+- Much like project [flak8](https://pypi.org/project/flake8/) where a [list of validations](https://www.flake8rules.com/) can be individually enabled/disabled, individual bundle validating rule should also be separable. Ultimately, tools that are interested in importing the validation library should be able to cherry-pick validations used for different scenarios and configurations. 
+
+- The library should provide all validations currently available within the operator-courier.
+
+- The bundle validation library will be used to validate OLM/registry manifests and expose an interface to add validators for any scheme. 
+
+- Errors and alerting messages should be formatted in a standardized fashion.
+
+- The structure of the operator-framework/api repository needs to be well-organized and defined in a Readme to encourage community contribution.
+
+- The library should allow customized validation strategies as part of an extended package.
+
+#### Initial set of validations
+
+There exist validation steatagies avaliable in _operator-courier_ and _api_ repositories which would serve as an initial set of validations:
+
+- Validate the metadata for every CRD that exists in a bundle.
+    - E01: crd metadata not defined
```

Comment:
I think our error codes should be unique across everything we validate (not a huge deal for the proposal, just a note for implementation)